### ISSUE TITLE
PSR-16 doesn't mention InvalidArgumentException for invalid TTL

### ISSUE THREAD

**Issue by matthiasmullie** [php-cache/integration-tests#75]
The current PSR-16 integrations tests expect an `InvalidArgumentException` to be thrown[1] when passing an invalid $ttl. I think that is wrong, or at least premature. The spec does not specify[2] what should happen in the case of invalid input and leaves that up to the library.

There is a PR[3] to make it official that an invalid TTL results in an `InvalidArgumentException`. However, it has not been merged in a month.

There is another PR[4] (to PSR-6, but I assume PSR-16 to follow suit should it pass) to throw a TypeError (or trigger_error). It's been around for longer, but could be brought back up with FIG 3 in place.

In short: I don't think this behavior for TTL should be enforced, since 1) it's undefined in the spec, and 2) there are competing ideas on how to eventually define it in spec.

1: https://github.com/php-cache/integration-tests/blob/master/src/SimpleCacheTest.php#L509
2: https://groups.google.com/forum/#!topic/php-fig/TQX0-DpKTKM
3: https://github.com/php-fig/fig-standards/pull/858
4: https://github.com/php-fig/fig-standards/pull/787

**Comment by nicolas-grekas** [comment]
Closing as explained in https://github.com/php-cache/integration-tests/pull/77#issuecomment-718464198
